### PR TITLE
✅ New decimal-aware price system - Part 10: Fix integration tests

### PR DESCRIPTION
**Pull request by JuaniRios**
Main info on #260

- This PR focuses on fixing the integration tests
- While fixing I noticed and fixed an issue where the CT migration logic was doing the funding check with a hardcoded value instead of the actual CT amount minted. (by funding check I refer to checking that the polimec sovereign account has enough tokens on the project chain).

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/272?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#272** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/272?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#269** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/269?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#270](https://github.com/Polimec/polimec-node/pull/270) <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/270?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#271](https://github.com/Polimec/polimec-node/pull/271) <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/271?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#268** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/268?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#267** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/267?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#266** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/266?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#265** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/265?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#264** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/264?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#263** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/263?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#262** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/262?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#260** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/260?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JuaniRios and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pallets/polimec-receiver/src/lib.rs [hunk 1]
```diff
@@ -110,6 +109,13 @@ pub mod pallet {
 				..
 			} in migrations.clone().inner()
 			{
+				pub const STANI: [u8; 32] = [
```
Comment (lrazovic) [inline pallets/polimec-receiver/src/lib.rs:112]: Remove the debug leftovers

File: integration-tests/src/constants.rs [hunk 2]
```diff
@@ -463,12 +463,12 @@ pub mod penpal {
 				),
 				INITIAL_DEPOSIT,
 			),
-			(
-				<Penpal<PolkadotNet>>::sovereign_account_id_of(
-					(Parent, xcm::prelude::Parachain(polimec::PARA_ID)).into(),
-				),
-				2_000_000_0_000_000_000,
-			), // i.e the CTs sold on polimec
+			// (
```
Comment (lrazovic) [inline integration-tests/src/constants.rs:466]: Remove

File: integration-tests/src/tests/e2e.rs [hunk 3]
```diff
@@ -347,25 +367,30 @@ fn remainder_round_completed() {
...
+		let funding_necessary_1 = inst.calculate_contributed_funding_asset_spent(excel_contributions(), price);
+		let funding_necessary_2 = inst.calculate_contributed_funding_asset_spent(excel_remainders(), price);
 		let mut total = 0u128;
 		for item in funding_necessary_1 {
 			total += item.asset_amount;
 		}
 		for item in funding_necessary_2 {
 			total += item.asset_amount;
 		}
+		dbg!(total);
```
Comment (lrazovic) [inline integration-tests/src/tests/e2e.rs:379]: Remove

File: integration-tests/src/tests/e2e.rs [hunk 4]
```diff
@@ -474,17 +515,18 @@ fn ct_migrated() {
 	// Migration is ready
 	PolitestNet::execute_with(|| {
 		let project_details = pallet_funding::ProjectsDetails::<PolitestRuntime>::get(project_id).unwrap();
+		dbg!(&project_details);
```
Comment (lrazovic) [inline integration-tests/src/tests/e2e.rs:518]: Remove

File: pallets/funding/src/functions.rs [hunk 5]
```diff
@@ -1791,11 +1793,17 @@ impl<T: Config> Pallet<T> {
...
+		ProjectsDetails::<T>::insert(project_id, project_details.clone());
+
+		let total_cts_minted = <T as Config>::ContributionTokenCurrency::total_issuance(project_id);
 
 		// * Send the migration query *
-		let expected_tokens: MultiAsset =
-			(MultiLocation { parents: 0, interior: Here }, 1_000_000_0_000_000_000u128).into(); // 1MM units for migrations
+		let expected_tokens: MultiAsset = (
+			MultiLocation { parents: 0, interior: Here },
+			total_cts_minted.into().saturating_add(T::ExistentialDeposit::get().into()),
```
Comment (lrazovic) [inline pallets/funding/src/functions.rs:1803]: Check if we actually need to take care of ED

**Review by lrazovic — APPROVED** [review]
Approved with few comments

**Comment by JuaniRios** [comment]
### Merge activity

* **Apr 30, 9:52 AM EDT**: @JuaniRios started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Polimec/polimec-node/272).

**Pull request merged**